alpha/apps/kaltura/modules/extwidget/actions/embedPlaykitJsAction.class.php

```diff
@@ -56,16 +58,41 @@ public function execute()
 
 		$this->initMembers();
 
+		$updateUiConfBundleCacheKey = true;
 		$bundleContent = $this->bundleCache->get($this->bundle_name);
 		$i18nContent = $this->bundleCache->get($this->bundle_i18n_name);
 		$extraModulesNames = unserialize($this->bundleCache->get($this->bundle_extra_modules_names));
 
 		if (!$bundleContent || $this->regenerate)
 		{
-			list($bundleContent, $i18nContent, $extraModulesNames) = kLock::runLocked($this->bundle_name, array("embedPlaykitJsAction", "buildBundleLocked"), array($this), 2, 30);
+			try
+			{
+				list($bundleContent, $i18nContent, $extraModulesNames, $updateUiConfBundleCacheKey) = kLock::runLocked($this->bundle_name, array("embedPlaykitJsAction", "buildBundleLocked"), array($this), 2, 30);
+			}
+			catch (kCoreException $ex)
+			{
+				switch ($ex->getCode())
+				{
+					case kCoreException::LOCK_TIMED_OUT:
```

Comment:
Does this case cover the 2 cases? case 1 - Bundle is in action and new requests are rejected - 5xx error, 
case 2 - Bundler failed / reset and returns 4xx error?

The only case that should return error is when the bundled content is missing some of the pieces and user gets explicit error in the response